esiclient/v1/node_network.py

```diff
@@ -0,0 +1,203 @@
+            if neutron_port_id:
+                neutron_port = neutron_client.get_port(neutron_port_id)
+                network_id = neutron_port.network_id
+                if not parsed_args.network or filter_network.id == network_id:
+                    network = neutron_client.get_network(network_id)
+                    if len(neutron_port.fixed_ips) > 0:
+                        fixed_ip = neutron_port.fixed_ips[0]['ip_address']
+                    data.append([node.name, port.address,
+                                 network.name, fixed_ip])
+            elif not parsed_args.network:
+                data.append([node.name, port.address, None, None])
+
+        return ["Node", "MAC Address", "Network", "Fixed IP"], data
+
+
+class Attach(command.ShowOne):
+    """Attach network to node"""
+
+    log = logging.getLogger(__name__ + ".Attach")
+
+    def get_parser(self, prog_name):
+        parser = super(Attach, self).get_parser(prog_name)
+        parser.add_argument(
+            "node",
+            metavar="<node>",
+            help=_("Name or UUID of the node"))
+        parser.add_argument(
+            "network",
+            metavar="<network>",
+            help=_("Name or UUID of the network"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)" % parsed_args)
+
+        node_uuid = parsed_args.node
+        network_uuid = parsed_args.network
+
+        ironic_client = self.app.client_manager.baremetal
+        neutron_client = self.app.client_manager.network
+
+        node = ironic_client.node.get(node_uuid)
+        network = neutron_client.find_network(network_uuid)
+
+        if node.provision_state == MANAGEABLE:
+            # adopt the node
+            node_update = []
+            node_revert = []
+            if 'image_source' not in node.instance_info:
+                temp_image = node.driver_info['deploy_ramdisk']
+                node_update.append({'path': '/instance_info/image_source',
+                                    'value': temp_image,
+                                    'op': 'add'})
+                node_revert.append({'path': '/instance_info/image_source',
+                                    'op': 'remove'})
+            if 'capabilities' not in node.instance_info:
+                node_update.append({'path': '/instance_info/capabilities',
+                                    'value': "{\"boot_option\": \"local\"}",
+                                    'op': 'add'})
+                node_revert.append({'path': '/instance_info/capabilities',
+                                    'op': 'remove'})
+
+            if len(node_update) > 0:
+                ironic_client.node.update(node_uuid, node_update)
+            ironic_client.node.set_provision_state(node_uuid, ADOPT)
+            if len(node_revert) > 0:
+                ironic_client.node.update(node_uuid, node_revert)
```

Comment:
Should the above be wrapped in a `try/finally` block to prevent changes from accidentally persisting if the `set_provision_state` call fails?  E.g.:

```
try:
            if len(node_update) > 0:
                ironic_client.node.update(node_uuid, node_update)
            ironic_client.node.set_provision_state(node_uuid, ADOPT)
finally:
            if len(node_revert) > 0:
                ironic_client.node.update(node_uuid, node_revert)
```

Reply:
Ah, makes sense! I'll update this.